v3/onesignal-metabox/onesignal-metabox.php

```diff
@@ -49,14 +59,18 @@ function onesignal_metabox($post)
   <div id="os_options">
     <label for="os_segment">Send to segment</label>
     <select name="os_segment" id="os_segment">
-      <option value="All">All</option>
-      <?php
-      for ($i = 0; $i < count($json->segments); $i++) {
-        $selected = isset($post->os_meta['os_segment']) && $post->os_meta['os_segment'] === $json->segments[$i]->name ? 'selected' : '';
-        echo '<option value="' . $json->segments[$i]->name . '"' . $selected . '>' . $json->segments[$i]->name . '</option>';
-      }
-      ?>
-    </select>
+    <option value="All">All</option>
+    <?php
+    if ($json && is_array($json->segments)) {
+        foreach ($json->segments as $segment) {
+            $selected = isset($post->os_meta['os_segment']) && $post->os_meta['os_segment'] === $segment->name ? 'selected' : '';
+            echo '<option value="' . esc_attr($segment->name) . '"' . $selected . '>' . esc_html($segment->name) . '</option>';
+        }
```

Comment:
Question: this might be overly cautious but do we need to add a check for if `$segment->name` is set?

Reply:
Doesn't hurt